doc-source/design-principles.rst

```diff
@@ -271,18 +271,27 @@ we've provided some helpful decorators and context managers that validate
 ``ttl`` for you and will ``ERROR`` the check if evidence ``ttl`` has expired
 prior to executing the check's logic.
 
-* ``with_raw_evidences`` decorator: Use this decorator on your check method
-  when you know the full path and name of your raw evidences.  The decorator
-  takes as arguments, the paths to your raw evidences as strings.  It also
-  passes the raw evidences to the decorated method in the form of method
-  arguments.
+* ``with_raw_evidences``, ``with_external_evidences`` decorators: Use these
+  decorators on your check method when you know the full path and name of
+  your raw or external evidences.  Each decorator takes as arguments, the
+  paths to your raw or external evidences as strings or as ``FromEvidence``
+  named tuples.  ``FromEvidence`` has ``path`` and ``ev_class`` (evidence class)
+  as attributes.  If the requested evidences pass TTL validation the evidences
+  are then passed along to the decorated method in the form of method arguments.
+  Use ``FromEvidence`` when dealing with sub-classed ``RawEvidence`` or
+  ``ExternalEvidence``, and you want the evidence provided to the decorated
+  method to be cast as that sub-classed evidence otherwise use a string path
+  and the evidence will be provided as either ``RawEvidence`` or ``ExternalEvidence``.
 
 ``@with_*_evidences`` usage example::
 
   ...
-  from compliance.evidence import with_raw_evidences
+  from compliance.evidence import FromEvidence, with_raw_evidences
   ...
-  @with_raw_evidence('foo/evidence_bar.json', 'foo/evidence_baz.json')
+  @with_raw_evidence(
+      FromEvidence('foo/evidence_bar.json', BarEvidence),
```

Comment:
You would need to import `BarEvidence` name in the example. Perhaps from an user module like `from .evidence import BarEvidence`.

Reply:
Fixed in https://github.com/ComplianceAsCode/auditree-framework/pull/80/commits/6c71069dd93a05230ce429fda162668c9576183a